core/src/kura.rs

```diff
@@ -168,6 +171,12 @@ impl Kura {
                 }
             };
 
+            if failed_to_store_block.is_some() {
+                let error = failed_to_store_block_reason.unwrap();
```

Comment:
```suggestion
                let error = failed_to_store_block_reason.expect("Already know that failure occurred");
```

Which is also why we might want to reconsider the structure. We're checking the failure condition twice.

Reply:
change accepted